src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -292,39 +177,31 @@ public override async Task<IEnumerable<string>> EnumerateVersions(PackageURL pur
         {
             try
             {
-                string? packageName = purl?.Name;
+                string? packageName = purl.Name;
+                string? packageVersion = purl.Version;
                 if (packageName == null)
                 {
                     return null;
                 }
-                HttpClient httpClient = CreateHttpClient();
-
-                string? content = await GetHttpStringCache(httpClient, $"{RegistrationEndpoint}{packageName.ToLowerInvariant()}/index.json", useCache);
-                return content;
-            }
-            catch (Exception ex)
-            {
-                Logger.Debug(ex, $"Error fetching NuGet metadata: {ex.Message}");
-                return null;
-            }
-        }
-        
-        public async Task<JsonElement?> GetVersionUriMetadata(string versionUri, bool useCache = true)
-        {
-            try
-            {
-                HttpClient httpClient = CreateHttpClient();
 
-                string? content = await GetHttpStringCache(httpClient, versionUri, useCache);
-                if (string.IsNullOrEmpty(content)) { return null; }
+                // If no package version provided, default to the latest version.
+                if (string.IsNullOrWhiteSpace(packageVersion))
+                {
+                    string latestVersion = (await EnumerateVersions(purl, useCache)).First();
```

Comment:
I see this latest version pattern in lots of places in this file - what do you think about a private helper method called `LatestVersion`? Maybe that functionality is useful enough to raise it into the parent type?

Reply:
I thought about doing that yes. I can mess around with it tomorrow.

Reply:
Resolved in new PR.